_pytest/config.py

```diff
@@ -990,11 +990,14 @@ def pytest_load_initial_conftests(self, early_config):
 
     def _initini(self, args):
         ns, unknown_args = self._parser.parse_known_and_unknown_args(args, namespace=self.option.copy())
-        r = determine_setup(ns.inifilename, ns.file_or_dir + unknown_args, warnfunc=self.warn)
+        rootdir = ns.rootdir if ns.rootdir else None
+        r = determine_setup(ns.inifilename, ns.file_or_dir + unknown_args, warnfunc=self.warn, rootdir_cmd_arg=rootdir)
         self.rootdir, self.inifile, self.inicfg = r
         self._parser.extra_info['rootdir'] = self.rootdir
         self._parser.extra_info['inifile'] = self.inifile
         self.invocation_dir = py.path.local()
+        if ns.rootdir:
+            self.invocation_dir = self.rootdir
```

Comment:
Why is this here? I'm not sure we should be changing the `invocation_dir`, it is supposed to contain the directory where pytest was invoked from, regardless of the `rootdir`.